src/metabase/models/app/graph.clj

```diff
@@ -0,0 +1,72 @@
+(ns metabase.models.app.graph
+  "Code for generating and updating the App permissions graph. App permissions
+  are based on the permissions of the app's collection."
+  (:require [clojure.data :as data]
+            [metabase.models.collection-permission-graph-revision :as c-perm-revision
+             :refer [CollectionPermissionGraphRevision]]
+            [metabase.models.collection.graph :as graph]
+            [metabase.models.permissions :as perms]
+            [metabase.models.permissions-group :as perms-group]
+            [metabase.models.setting :as setting :refer [defsetting]]
+            [metabase.public-settings.premium-features :as premium-features]
+            [metabase.util.i18n :as i18n :refer [tru]]
+            [metabase.util.schema :as su]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+(def ^:private AppPermissions
+  graph/CollectionPermissions)
+
+(def ^:private GlobalPermissionsGraph
+  {:revision s/Int
+   :groups   {su/IntGreaterThanZero AppPermissions}})
+
+(s/defn ^:private set-all-users-app-permission!
+  [permission :- AppPermissions]
+  (setting/set-value-of-type! :keyword :all-users-app-permission permission))
+
+(defsetting all-users-app-permission
+  "App permission of the All Users group"
+  :type :keyword
+  :visibility :internal
+  :default :none
+  :setter (fn [v]
+            (set-all-users-app-permission (cond-> v (string? v) keyword))))
+
+(defn set-default-permissions!
+  "Sets the default permissions for the ''All Users'' group on`app` as specified
+  by `all-users-app-permission` if advanced permissions are not available."
+  [app]
+  (when-not (premium-features/has-feature? :advanced-permissions)
```

Comment:
As discussed in our 1:1: it might make sense to make this an assertion if it's really not meant to be used in EE rather than no-op